mrmustard/lab/representations/data/gaussian_data.py

```diff
@@ -0,0 +1,175 @@
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+import numpy as np
+#from numba import njit
+from typing import Optional, Union
+from mrmustard.lab.representations.data import MatVecData
+from mrmustard.math import Math
+from mrmustard.typing import Batch, Matrix, Scalar, Vector
+
+math = Math()
+
+
+class GaussianData(MatVecData):
+
+    def __init__(
+        self,
+        cov: Optional[Batch[Matrix]] = None,
+        mean: Optional[Batch[Vector]] = None,
+        coeffs: Optional[Batch[Scalar]] = None,
+    ) -> None:
+        r"""
+        Gaussian data: covariance, mean, coefficient.
+        Each of these has a batch dimension, and the length of the
+        batch dimension is the same for all three.
+        These are the parameters of a linear combination of Gaussians,
+        which is Gaussian if there is only one contribution for each.
+        Each contribution parametrizes the Gaussian function:
+        `coeffs * exp(-0.5*(x-mean)^T cov^-1 (x-mean))`.
+        Args:
+            cov (batch, dim, dim): covariance matrices (real symmetric)
+            mean  (batch, dim): means (real)
+            coeffs (batch): coeffsicients (complex)
+        """
+    
+        if (cov or mean) is not None:
+    
+            if cov is None:
+                dim = mean.shape[-1]
+                batch_size = mean.shape[-2]
+                cov = math.astensor([math.eye(dim, dtype=mean.dtype) for _ in range(batch_size)])
+
+            else: # we know mean is None here
+                dim = cov.shape[-1]
+                batch_size = cov.shape[-3]
+                mean = math.zeros((batch_size, dim), dtype=cov.dtype)
+        else:
+            raise ValueError("You need to define at one: covariance or mean")
+
+        if coeffs is None:
+            batch_size = cov.shape[-3]
+            coeffs = math.ones((batch_size), dtype=mean.dtype)
+
+        # TODO : decide on what this is, whether we keep it do we keep it ?
+        if isinstance(cov, QuadraticPolyData):  # enables GaussianData(quadraticdata)
+            poly = cov  # for readability
+            inv_A = math.inv(poly.A)
+            cov = 2 * inv_A
+            mean = 2 * math.solve(poly.A, poly.b)
+            coeffs = poly.c * math.cast(
+                math.exp(0.5 * math.einsum("bca,bcd,bde->bae", mean, cov, mean)), poly.c.dtype
+            )
+
+        else: # why else, isn't this just part of the standard init???
+            super().__init__(mat=cov, vec=mean, coeffs=coeffs)
+
+
+        
+    @property
+    def cov(self) -> Batch[Matrix]:
+        return self.mat
+
+
+
```

Comment:
See comment above. Essentially there are two modes of operation: `GaussianData(cov, means, coeff)` and `GaussianData(qdata)`. If you prefer, we can put that code in a classmethod so we can do `GaussianData.from_QData(qdata)`.

Reply:
Made it a classmethod. cheers!